import/CoordinateConversionController.qml

```diff
@@ -0,0 +1,277 @@
+        }
+        // Point has changed, update!
+        currentPointChanged(internal.currentPoint);
+    }
+
+    /*!
+    * \brief Given option, generates a new result.
+    * 
+    * A new CoordianteConversionResult is added to the list-model returned by
+    * coordinateResults. This Result is tied to the conversion parameters as
+    * given by option.
+    * 
+    * \param option option to generate result for.
+    */
+    function addNewCoordinateResultForOption(option) {
+        let result = internal.resultBuilder.createObject(this, { type: option });
+        currentPointChanged.connect(result.updateCoordinatePoint);
+        result.updateCoordinatePoint(internal.currentPoint);
+        internal.results.push(result);
+        // Regenerate results list.
+        results = internal.results;
+    }
+
+    /*!
+    * \brief Removes a given CoordinateConversionResult at index.
+    * 
+    * This function is for deleting results in the list-model returned by
+    * coordinateREsults.
+    * 
+    * \param index index of a given CoordinateConversionResult in the list-model
+    * returned by coordinateResults.
+    */
+    function removeCoordinateResultAtIndex(index) {
+        internal.results.splice(index, 1);
+        // Regenerate results list.
+        results = internal.results;
+    }
+
+    /*!
+    * \brief Updates the GeoView's camera to point to the current point's
+    * location on the map.
+    * \sa zoomToDistance
+    */
+    function zoomToCurrentPoint() {
+        if (geoView === null)
+            return;
+
+        if (geoView.currentViewpointCamera !== undefined) {
+            // This is a SceneView.
+            let camera = geoView.currentViewpointCamera;
+            let newCamera = ArcGISRuntimeEnvironment.createObject("Camera", {
+                "location": internal.currentPoint,
+                "distance": zoomToDistance,
+                "heading": camera.heading,
+                "roll": camera.roll,
+                "pitch": camera.pitch
+
+            });
+            geoView.setViewpointCameraAndSeconds(newCamera, 1.0);
+        }
+        else if (geoView.mapRotation !== undefined) {
+            // This is a MapView.
+            let newViewpoint = ArcGISRuntimeEnvironment.createObject(
+                "ViewpointCenter", {
+                "center": internal.currentPoint,
+                "targetScale": geoView.currentViewpointCenter.targetScale
+            });
+            geoView.setViewpointAndSeconds(newViewpoint, 1.0);
+        }
+    }
+
+    /*!
+    * \internal
+    * \brief This is a helper function to keep parity with the C++
+    * CoordinateConversionController. That controller can not pass `Point` 
+    * through QML to other controller objects.
+    * So this forces _that_ controller to emit the currentPointChanged
+    * signal again: which force all downstream consumers to update.
+    * 
+    * If a C++ point were consumable in QML then we would just call the point
```

Comment:
so this can all go away with api registration of Point?

Reply:
It should do!